### PR TITLE
imitate same behaviour as std::filesystem::is_regular_file

### PR DESCRIPTION
**Pull request by pamapa**

### CODE REVIEW

File: src/util/tools.cc [hunk 1]
```diff
@@ -176,9 +176,24 @@ bool isRegularFile(const fs::path& path)
     // unfortunately fs::is_regular_file(path, ec) does not to work for files >2GB on ARM 32bit systems (see #737)
     struct stat statbuf;
     int ret = stat(path.c_str(), &statbuf);
-    if (ret != 0)
+    if (ret != 0) {
+        throw std::runtime_error(mt_strerror(errno) + ": " + path.string());
```
Comment (whyman) [inline src/util/tools.cc]: Maybe we should include the function name in the error, just to make it 1000% clear?

Comment (pamapa) [reply src/util/tools.cc]: good point will do i follow up

Comment (pamapa) [reply src/util/tools.cc]: hmmm there multiple throws in tools.cc, etc

Should be do something like:
```
#define throw_runtime_error(msg) throw std::runtime_error(std::string(msg) \
  + " line: " + std::to_string(__LINE__) + " file: " + std::string(__FILE__))
```
in exceptions.h
?

Comment (whyman) [reply src/util/tools.cc]: Yes, nice idea!

**Pull request merged**